test/MrkdwnJsFileType.test.js

```diff
@@ -0,0 +1,173 @@
+/*
+ * MrkdwnJsFileType.test.js - test the Mrkdwn file type handler object.
+ *
+ * Copyright © 2024 Box, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+var fs = require("fs");
+if (!MrkdwnJsFileType) {
+    var MrkdwnJsFileType = require("../MrkdwnJsFileType.js");
+    var CustomProject =  require("loctool/lib/CustomProject.js");
+    var ProjectFactory =  require("loctool/lib/ProjectFactory.js");
+}
+var p = new CustomProject({
+    sourceLocale: "en-US",
+    plugins: ["../."]
+}, "./test/testfiles", {
+    locales:["en-GB"],
+    mrkdwn: {
+        "mappings": {
+            "file.js": {
+                "template": "resources/[localeDir]/file.js"
+            },
+            "resources/en/US/file.js": {
+                "template": "resources/[localeDir]/file.js"
+            },
+            "**/messages.js": {
+                "template": "resources/[localeDir]/messages.js"
+            },
+            "**/test/str.js": {
+                "template": "[dir]/[localeDir]/str.js"
+            },
+            "**/*.js": {
+                "template": "[localeDir]/[filename]"
+            }
+        }
+    }
+});
+var p2 = new CustomProject({
+    sourceLocale: "en-US"
+}, "./test/testfiles", {
+    locales:["en-GB"],
+    flavors: ["ASDF"],
+    mrkdwn: {
+    }
+});
+describe("mrkdwnfiletype", function() {
+    test("MrkdwnJsFileTypeConstructor", function() {
```

Comment:
What do you think of updating test case name, so that it is self explanatory?
So instead 
```test("MrkdwnJsFileTypeConstructor", () => {})```
 we do
```test("", (creates MrkdwnJsFileType instance) => {})```

Reply:
These unit test names were all copied from ilib-loctool-ghfm (the github flavored markdown plugin for loctool), which was way easier than starting from scratch. These, in turn, can from the original nodeunit  unit tests. Nodeunit required named functions, so you couldn't have really descriptive names/descriptions in English with properly spacing and punctuation, etc. When we converted these to jest with an automated process, we just kept the names because it was difficult to fix them nicely. There are thousands and thousands of these test cases we converted to jest, so we just decided that it was too much work and we will add nice spacing and punctuation when we write new tests.

Reply:
Thank you for sharing the context!
I noticed that all of these tests are new, so I thought it might be a good idea to change the convention and use more descriptive test names. In my experience, having clear and well-formatted test names can greatly enhance the readability and maintainability of our codebase.